### ISSUE TITLE
404 for mac GPU

### ISSUE THREAD

**Issue by tivvit** [tensorflow/tensorflow#11761]
Installation section in readme for Mac GPU: 404 error

https://ci.tensorflow.org/view/Nightly/job/nightly-matrix-mac-gpu/TF_BUILD_IS_OPT=OPT,TF_BUILD_IS_PIP=PIP,TF_BUILD_PYTHON_VERSION=PYTHON2,label=gpu-mac/lastSuccessfulBuild/artifact/pip_test/whl/tensorflow_gpu-1.3.0rc0-py2-none-any.whl

**Comment by reedwm** [comment]
Can you clarify? Where did you get that link?

**Comment by rafaelsimonmaia** [comment]
Not OP. But the Mac GPU Links are in the Installation section of the README.md

**Comment by reedwm** [comment]
Ah, thank you for clarifying.

@av8ramit, can you check why the Mac GPU links are broken?

**Comment by av8ramit** [comment]
Unfortunately, we stopped Mac GPU support after the 1.1 release. The build itself is broken as well and we have disabled it. We are accepting patches for Mac GPU, but no longer offer any official support. Sorry about that.

As for the links in the README, thanks for pointing that out! I've created a PR: https://github.com/tensorflow/tensorflow/pull/11790

**Comment by subzerofun** [comment]
@av8ramit  That's really unfortunate – i recently got myself a 1080 Ti and thought that with the 11 GB VRAM i can finally run all the stuff my older card couldn't lift. And i have just discovered Tensorboard ... Now i can only:
1) Downgrade >= 1.2 TF only functions/code back to 1.1 (won't work for long 😬)
2) Search for implementations of a TF project in Keras (Theano), pyTorch/Torch etc.
3) Rewrite TF only projects from scratch with another framework
4) Put the GPU in a separate Machine with Linux ...
5) Sell my body and/or organs so i can afford a 128 Core Xeon Machine

I currently have a working 1.1 GPU install – can i just make a copy of all files in my python `site-packages/tensorflow`  folder if i want to try out Tensorflow >=1.2 (CPU only ofc) and then revert back?

Have you seen if someone has managed to build TF 1.2 with GPU support from source? What is the biggest obstacle/time consuming issue that forced the TF team to drop Mac GPU support?

**Comment by rafaelsimonmaia** [comment]
I compiled TF 1.2 with GPU support for my mac pro. It runs fine so far. R1.3 is not really out yet, and it is a bit broken right now for macs. Biggest issue seems compiler compatibility. llvm/clang 9 is not really getting along with cuda 8 yet and some other stuff seem to be broken too.

**Comment by subzerofun** [comment]
@rafaelsimonmaia Cool! Could you point out what problems i could face when building from source – how to avoid getting stuck in the process? What did you need to install or change that you could compile it successfully?

**Comment by rafaelsimonmaia** [comment]
@subzerofun Perhaps you wanna give a try https://github.com/rafaelsimonmaia/tensorflow_for_macpro/blob/master/Compute%206.1/tensorflow-1.2.1-cp35-cp35m-macosx_10_7_x86_64.whl
It is for python 3 and Compute 6.1, so it should work on your machine. 

To compile, I downgraded my clang to Apple LLVM version 8.0.0 (clang-800.0.42.1), which is found on the Command Line Tools for Xcode 8.2. Then I updated Cuda to 8, and got the latest cuDNN 6 as well. I configure and said no to everything, save for the CUDA option.  Then I removed -lgomp from the `"third_party/gpus/cuda/BUILD.tpl"` file, as it doesn't seem to be needed for linking. Then `bazel build --config=opt --config=cuda //tensorflow/tools/pip_package:build_pip_package`  and `bazel-bin/tensorflow/tools/pip_package/build_pip_package /tmp/tensorflow_pkg`. 

I even got the current master to work as well, but for that I had to edit one variant.h file and remove a constexpr from a construction, as clang was breaking at that point. I may make a PR later to address that, if I find some time.

**Comment by subzerofun** [comment]
@rafaelsimonmaia Thanks for the links! Unfortunately there is only a python 3 install in the Compute 6.1 folder. Do you maybe have build a .whl file for python 2.7 too? I have two cards with Compute Abilities: **3.5** (gtx 780) and **6.1** (gtx 1080 ti). So unfortunately i can't use your python 2.7 file (it's for Compute 5.2).

Do you maybe also have a working .whl file for TF 1.3 (and some instructions how to prepare other files and configurations) ?

I'm currently still on 1.1 (which works fine on most projects) with GPU support. Would love to upgrade (if it was as easy as before...).

**Comment by sstolpovskiy** [comment]
@subzerofun I was able to compile master with removing constexpr in variant.h
` class Variant {
  public:
-  constexpr Variant() noexcept = default;
+  Variant() noexcept = default;
`
It looks working, but I'm wondering if it's broke something?
